src/omv/modules/py_tf.c

```diff
@@ -629,7 +623,49 @@ STATIC void py_tf_segment_output_data_callback(void *callback_data,
     }
 }
 
-STATIC mp_obj_t int_py_tf_segment(bool detecting_mode, uint n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+STATIC mp_obj_t py_tf_segment(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_roi };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_roi, MP_ARG_OBJ | MP_ARG_KW_ONLY, {.u_rom_obj = MP_ROM_NONE} }
+    };
+
+    // Parse args.
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 2, pos_args + 2, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    image_t *image = py_helper_arg_to_image(pos_args[1], ARG_IMAGE_ANY);
+    rectangle_t roi = py_helper_arg_to_roi(args[ARG_roi].u_obj, image);
+
+    fb_alloc_mark();
+    py_tf_alloc_putchar_buffer();
+
+    py_tf_model_obj_t *model = py_tf_load_alloc(pos_args[0]);
+    uint8_t *tensor_arena = fb_alloc(model->params.tensor_arena_size, FB_ALLOC_PREFER_SPEED | FB_ALLOC_CACHE_ALIGN);
+
+    py_tf_input_data_callback_data_t py_tf_input_data_callback_data;
+    py_tf_input_data_callback_data.img = image;
+    py_tf_input_data_callback_data.roi = &roi;
+
+    mp_obj_t py_tf_segment_output_data_callback_data;
+
+    if (libtf_invoke(model->model_data,
+                     tensor_arena,
+                     &model->params,
+                     py_tf_input_data_callback,
+                     &py_tf_input_data_callback_data,
+                     py_tf_segment_output_data_callback,
+                     &py_tf_segment_output_data_callback_data) != 0) {
```

Comment:
I think those variables/functions can be shortened out and made more readable, for example:

```suggestion
                     py_tf_input_callback,
                     &py_tf_input_callback_data,
                     py_tf_segment_output_callback,
                     &py_tf_segment_output_callback_data) != 0) {
```
And the same goes for classify.

Reply:
Done

Reply:
Did you forgot to push the changes ?

Reply:
About to.